### PR TITLE
Improving create graph

### PR DESCRIPTION
**Pull request by mmatera**
Here I started to clean up the routine that creates graphs. It is a spaghetti mess, making it harder to figure out what it was it is supposed to do. I guess that fixing this routine several doctest would start passing. Actually, with these few changes, two more tests are passing now.

Also, I have incorporated the changes in #9. However,  it seems that there is a problem setting ```MATHICS_CHARACTER_ENCODING``` in pytest from here.

### CODE REVIEW

File: pymathics/graph/base.py [hunk 1]
```diff
@@ -725,14 +722,16 @@ def _graph_from_list(rules, options, new_vertices=None):
             new_edges, new_edge_properties, options=options, new_vertices=new_vertices
         )
 
-
+
```
Comment (mmatera) [inline pymathics/graph/base.py:725]: This is the only function that I actually was working on. Other changes come from #9.

File: Makefile [hunk 2]
```diff
@@ -54,7 +52,7 @@ clean-pyc:
 
 #: Run py.test tests. Use environment variable "o" for pytest options
 pytest:
-	$(PYTHON) -m pytest test $o
+	MATHICS_CHARACTER_ENCODING="ASCII" $(PYTHON) -m pytest $(PYTEST_WORKERS) test $o
```
Comment (mmatera) [inline Makefile:55]: For some reason, this does not work here...

Comment (mmatera) [reply Makefile:55]: Now it does

File: pymathics/graph/__init__.py [hunk 3]
```diff
@@ -118,6 +121,7 @@
     "GraphAtlas",
     "GraphAtom",
     "GraphBox",
+    "GraphData",
```
Comment (mmatera) [inline pymathics/graph/__init__.py:124]: Since we use "__all__" we need to list all the symbols here

File: test/helper.py [hunk 4]
```diff
@@ -4,12 +4,10 @@
 
 from mathics.session import MathicsSession
 
-session = MathicsSession(add_builtin=True, catch_interrupt=False)
-
-# Set up a Mathics3 session with definitions.
+# Set up a Mathics session with definitions.
 # For consistency set the character encoding ASCII which is
 # the lowest common denominator available on all systems.
-session = MathicsSession(character_encoding="ASCII")
```
Comment (mmatera) [inline test/helper.py:12]: Now we create the builtin symbols  once, so the parameter in this second call does not have any effect.

**Comment by mmatera** [comment]
With these changes, 8 doctests more pass now.

File: pymathics/graph/base.py [hunk 5]
```diff
@@ -1566,7 +1603,7 @@ class KatzCentrality(_ComponentwiseCentrality):
     """
 
     rules = {
-        "KatzCentrality[g_, alpha_]": "KatzCentrality[g, alpha, 1]",
+        "Pymathics`KatzCentrality[g_, Pymathics`alpha_]": "Pymathics`KatzCentrality[g, Pymathics`alpha, 1]",
```
[the remark this answers is not in the document]
Comment (rocky) [reply pymathics/graph/base.py:1606]: And even this is not quite right. See above.

**Comment by rocky** [comment]
LGTM - see comments though.

File: pymathics/graph/base.py [hunk 6]
```diff
@@ -423,7 +423,7 @@ class _FullGraphRewrite(Exception):
 
 def _normalize_edges(edges):
     for edge in edges:
-        if edge.has_form("Property", 2):
+        if edge.has_form("Pymathics`Property", 2):
```
[the remark this answers is not in the document]
Comment (rocky) [reply pymathics/graph/base.py:426]: BTW, the context should be added to Symbols defined in `pymathics.graph.graphsymbols`.

File: pymathics/graph/base.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (mmatera) [reply pymathics/graph/base.py:1606]: See #11

**Pull request merged**